src/update_agent/actor.rs

```diff
@@ -120,15 +119,21 @@ impl UpdateAgent {
     /// This influences the pace of the update-agent refresh loop. Timing of the
     /// state machine is not uniform. Some states benefit from more/less
     /// frequent refreshes, or can be customized by the user.
-    fn refresh_delay(&self) -> Duration {
-        let default_delay = Duration::from_secs(super::DEFAULT_REFRESH_PERIOD_SECS);
+    fn refresh_delay(&self, prev_state: UpdateAgentState) -> Option<Duration> {
+        use std::mem::discriminant;
+
+        // State changes trigger immediate tick/action.
+        if discriminant(&prev_state) != discriminant(&self.state) {
```

Comment:
Huh neat, hadn't seen `discriminant()` before.

Reply:
I'm honestly still looking forward for https://github.com/rust-lang/rust/issues/60553 to be implemented and stabilized.